app/models/application_record.rb

```diff
@@ -3,4 +3,21 @@
 # Abstract record class to be applied to all models
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
+
+  def self.arel_find(collection, batch_size = 1000)
+    base_arel = arel_table[column_names.first].eq('')
+    collection.in_groups_of(batch_size, false).map do |batch|
+      where(batch.inject(base_arel) do |rel, record|
+        rel.or(yield(record))
+      end)
+    end.inject([], &:+)
+  end
```

Comment:
This replaces an old paradigm that doesn't perform quite as well:

```rb
models.inject(where('1=0')) do |rel, model|
      rel.or(where(foo: model.foo, bar: model.bar))
end
```

By using arel directly instead of AR, we have more room for large queries before hitting a `SystemStackError` when the stack grows too large. The old way allowed up to about 200 records in the collection; the new code allows about 2000 records. I've set the default batch size to 1000 to be safe, and we can reduce it further in the future if it becomes problematic.